--- tuareg.el.orig	2006-01-09 10:33:32.000000000 +0100
+++ tuareg.el		2006-01-09 10:35:01.000000000 +0100
@@ -881,7 +881,7 @@
 (defvar tuareg-font-lock-keywords
   (append
   (list
-   (list "\\<\\(external\\|open\\|include\\|rule\\|s\\(ig\\|truct\\)\\|module\\|functor\\|with[ \t\n]+\\(type\\|module\\)\\|val\\|type\\|method\\|virtual\\|constraint\\|class\\|in\\|inherit\\|initializer\\|let\\|rec\\|and\\|begin\\|object\\|end\\)\\>"
+   (list "\\<\\(external\\|open\\|include\\|rule\\|s\\(ig\\|truct\\)\\|module\\|functor\\|with[ \t\n]+\\(type\\|module\\)\\|val\\|type\\|method\\|virtual\\|constraint\\|class\\|in\\|inherit\\|initializer\\|perform\\|let\\|rec\\|and\\|begin\\|object\\|end\\)\\>"
 	 0 'tuareg-font-lock-governing-face nil nil)
    (list "\\.<\\|>\\.\\|\\.~\\|\\.!"
 	 0 'tuareg-font-lock-multistage-face nil nil)
@@ -1263,7 +1263,7 @@
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;;                               Indentation stuff
 
-(defconst tuareg-keyword-regexp "\\<\\(object\\|initializer\\|and\\|c\\(onstraint\\|lass\\)\\|m\\(atch\\|odule\\|ethod\\|utable\\)\\|s\\(ig\\|truct\\)\\|begin\\|e\\(lse\\|x\\(ception\\|ternal\\)\\)\\|t\\(o\\|hen\\|ry\\|ype\\)\\|v\\(irtual\\|al\\)\\|w\\(h\\(ile\\|en\\)\\|ith\\)\\|i\\(f\\|n\\(herit\\)?\\)\\|f\\(or\\|un\\(ct\\(or\\|ion\\)\\)?\\)\\|let\\|do\\(wnto\\)?\\|parser?\\|rule\\|of\\)\\>\\|->\\|[;,|]"
+(defconst tuareg-keyword-regexp "\\<\\(object\\|initializer\\|and\\|c\\(onstraint\\|lass\\)\\|m\\(atch\\|odule\\|ethod\\|utable\\)\\|s\\(ig\\|truct\\)\\|begin\\|e\\(lse\\|x\\(ception\\|ternal\\)\\)\\|t\\(o\\|hen\\|ry\\|ype\\)\\|v\\(irtual\\|al\\)\\|w\\(h\\(ile\\|en\\)\\|ith\\)\\|i\\(f\\|n\\(herit\\)?\\)\\|f\\(or\\|un\\(ct\\(or\\|ion\\)\\)?\\)\\|let\\|do\\(wnto\\)?\\|parser?\\|perform\\|rule\\|of\\)\\>\\|->\\|[;,|]"
   "Regexp for all recognized keywords.")
 
 (defconst tuareg-match-|-keyword-regexp
@@ -1332,6 +1332,7 @@
     ("mutable" . tuareg-default-indent)
     ("external" . tuareg-default-indent)
     ("in" . tuareg-in-indent)
+    ("perform" . tuareg-default-indent)
     ("of" . tuareg-default-indent)
     ("to" . tuareg-default-indent)
     ("downto" . tuareg-default-indent)
@@ -1382,7 +1383,7 @@
     (if found kwop (goto-char (point-min)) nil)))
 
 (defconst tuareg-find-kwop-regexp
-  (concat tuareg-matching-keyword-regexp "\\|\\<\\(for\\|while\\|do\\|if\\|begin\\|s\\(ig\\|truct\\)\\|object\\)\\>\\|[][(){}]\\|\\.<\\|>\\.\\|\\*)"))
+  (concat tuareg-matching-keyword-regexp "\\|\\<\\(for\\|while\\|do\\|if\\|begin\\|s\\(ig\\|truct\\)\\|object\\|perform\\)\\>\\|[][(){}]\\|\\.<\\|>\\.\\|\\*)"))
 (defun tuareg-make-find-kwop-regexp (kwop-regexp)
   (concat tuareg-find-kwop-regexp "\\|" kwop-regexp))
 
